sensortag20/BleSensorTag/src/main/java/com/example/ti/ble/sensortag/ButtonPressListener.java

```diff
@@ -0,0 +1,98 @@
+/**************************************************************************************************
+ * author: Ian Parfitt
+ * file: ButtonPressListener.java
+ *
+ * This class handles determining how long button presses have happened for, and builds an array of
+ * button press data, to be parsed as a button pattern for unlocking a door.  Packages results as
+ * an array
+ */
+package com.example.ti.ble.sensortag;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Date;
+import com.example.ti.util.KnockTrendsCustomTimer;
+
+public class ButtonPressListener {
+    private KnockTrendsCustomTimer mTimer;
+    private static ButtonPressListener instance = null;
+    private List<Long> mList;
+    private long currentMilliseconds;
+    private long lastMilliseconds;
+
+    // Singleton protected constructor.  NO ONE GETS IN.
+    private ButtonPressListener() {
+        mTimer = new KnockTrendsCustomTimer(2, this); // 2 second timeout
+        mList = new ArrayList<Long>();
+        currentMilliseconds = 0;
+        lastMilliseconds = 0;
+    }
+
+    // Publicly available "constructor"
+    public static ButtonPressListener getInstance() {
+        if(instance == null) {
+            instance = new ButtonPressListener();
+        }
+        return instance;
+    }
+
+    private long buttonOff() {
+        Date date = new Date();
+        long milliseconds = date.getTime();
+        mTimer.start();
+        return milliseconds;
+    }
+
+    private long buttonOn() {
+        Date date = new Date();
+        long milliseconds = date.getTime();
+        mTimer.stop();
+        return milliseconds;
+    }
+
+    private Long[] resetValues() {
+
+        // Still not sure why the first value is useless, but this works. ¯\_(ツ)_/¯
```

Comment:
@JakeDarling I didn't spend any more time sorting it out. It seems consistent that there is an extra value at the beginning of the list, so just removing that value works for now. If we run into problems in the future, we can refactor and find the issue, but it didn't seem worth the time for now.